presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeSearcher.java

```diff
@@ -125,7 +125,7 @@ public PlanNodeSearcher skipOnlyWhen(Predicate<PlanNode> skipOnly)
         if (where.test(node)) {
             nodes.add((T) node);
         }
-        if (skipOnly.test(node)) {
+        if (skipOnly.test(node) || where.test(node)) {
```

Comment:
`skipOnly` is named terribly wrong. However as long as it means "continue search only when", you should _not_ modify this `if`.